modules/peridynamics/src/kernels/HorizonStabilizedFormIIFiniteStrainMechanicsNOSPD.C

```diff
@@ -49,7 +49,7 @@ HorizonStabilizedFormIIFiniteStrainMechanicsNOSPD::computeLocalResidual()
                       ((_dgrad[nd].det() * _stress[nd] * _dgrad[nd].inverse().transpose()) *
                        _shape2[nd].inverse())
                           .row(_component) *
-                      _origin_vec * _bond_status;
+                      _origin_vec * _node_vol[1 - nd] * _bond_status;
```

Comment:
It might not be a bad idea to add something like this at the beginning of this function:
```
mooseAssert(_nnodes < 2, "Function only works with 2 or fewer nodes per element");
```
This would make sure you never end up with a negative index into `_node_vol`.

Reply:
__nnodes_ is a protected parameter of the parent class. Its value is 2.

Reply:
Sure, but the point of an assertion is to make sure that your assumptions of its value are actually true. The assertion only gets executed for the debug executable, so it doesn't hurt performance.

Reply:
In looking at this again, I think my suggestion is a bad idea.